dap-mode.el

```diff
@@ -1598,6 +1606,27 @@ point is set."
   (interactive)
   (dap-mode t))
 
+
+
+;; Auto configure
+
+(define-minor-mode dap-auto-configure-mode
+  "Auto configure dap minor mode."
+  :init-value dap-auto-configure
+  :group 'dap-mode
+  (cond
+   (dap-auto-configure-mode
+    (dap-ui-mode 1)
+    (when (require 'posframe nil t)
+      (dap-ui-controls-mode 1))
+    (add-hook 'dap-stopped-hook #'dap-ui-show-debug-windows)
+    (add-hook 'dap-terminated-hook #'dap-ui-hide-debug-windows))
```

Comment:
I would put this in a separate mode so it could be turned off and on independently.